### PR TITLE
type: add db prefix to function name for not exist error messages

### PR DESCRIPTION
**Pull request by lizhenda**
### What problem does this PR solve? 
Close https://github.com/pingcap/tidb/issues/11147

### What is changed and how it works?
add db prefix to function name for not exist error messages

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

**Comment by winoros** [comment]
@lizhenda CI failed. Please take a look.

**Comment by lizhenda** [comment]
> @lizhenda CI failed. Please take a look.

OK

**Comment by lizhenda** [comment]
> @lizhenda CI failed. Please take a look.

CI success

**Comment by lizhenda** [comment]
@winoros 
CI success

**Comment by SunRunAway** [comment]
Friendly ping, @lizhenda

**Comment by lizhenda** [comment]
@SunRunAway Done, please have a look.

File: expression/integration_test.go [hunk 1]
```diff
@@ -4611,3 +4611,13 @@ func (s *testIntegrationSuite) TestDatetimeMicrosecond(c *C) {
 		testkit.Rows("2007-03-28 22:08:27.999998"))
 
 }
+
+func (s *testIntegrationSuite) TestNotExistFunc(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	_, err := tk.Exec("SELECT xxx(1)")
+	c.Assert(err.Error(), Equals, "[expression:1305]FUNCTION xxx does not exist")
```
Comment (XuHuaiyu) [inline expression/integration_test.go]: What will MySQL return if database is empty?

Comment (lizhenda) [reply expression/integration_test.go]: MySQL returns "ERROR 1046 (3D000): No database selected".  I will fix this in this PR.

File: expression/scalar_function.go [hunk 2]
```diff
@@ -80,7 +80,12 @@ func newFunctionImpl(ctx sessionctx.Context, fold bool, funcName string, retType
 	}
 	fc, ok := funcs[funcName]
 	if !ok {
-		return nil, errFunctionNotExists.GenWithStackByArgs("FUNCTION", funcName)
+		dbFuncName := ctx.GetSessionVars().CurrentDB
+		if dbFuncName != "" {
+			dbFuncName += "."
+		}
```
Comment (zz-jason) [inline expression/scalar_function.go]: if `dbFuncName == ""`, we should return an `error` to indicate that `ERROR 1046 (3D000): No database selected`.

Comment (zz-jason) [reply expression/scalar_function.go]: @lizhenda Please take a look at this comment.

Comment (lizhenda) [reply expression/scalar_function.go]: fix in the latest commit.

Comment (SunRunAway) [reply expression/scalar_function.go]: Could you add one more test case covering`No database selected`

Comment (lizhenda) [reply expression/scalar_function.go]: Add test case for function without params

**Comment by lizhenda** [comment]
@zz-jason integration-common-test failed

**Comment by SunRunAway** [comment]
> It seems that, not for sure, we failed to cherry-pick this commit to release-2.1. Please comment '/run-cherry-picker' to try to trigger the cherry-picker if we did fail to cherry-pick this commit before. @SunRunAway PTAL.

It's already picked.

**Pull request merged**